### PR TITLE
Document elementwise functions for DOK arrays

### PR DESCRIPTION
**Pull request by DragaDoncila**
@hameerabbasi I've updated the docs to include elementwise operations for DOK arrays. 

At this point I'm not sure whether you want to keep the DOK documentation in Construct Sparse Arrays, or whether you'd like to pull it out as its own section, but please let me know if that's the case, and whether you want more detail on the elementwise operations e.g. more detail on comparisons, etc. As it is, I tried to follow the convention already used for the COO arrays.

### CODE REVIEW

**Review by hameerabbasi — COMMENTED** [review]
It'd be nice to note all standard ufuncs and operators are now supported on DOK minus reduce.

File: docs/construct.rst [hunk 1]
```diff
@@ -140,8 +140,31 @@ DOK arrays also support fancy indexing assignment if and only if all dimensions
...
+to any number of arguments where the arguments can be :obj:`SparseArray` objects, :obj:`scipy.sparse.spmatrix`
+objects, or :obj:`numpy.ndarrays`. 
+
+.. code-block:: python
+
+   x = sparse.random((10, 10), 0.5, format="dok")
+   y = sparse.random((10, 10), 0.5, format="dok")
+   sparse.elemwise(np.add, x, y)
+
+:obj:`DOK` arrays also support standard `__ufunc__` operations and operators, including comparison operators,
```
Comment (hameerabbasi) [inline docs/construct.rst:153]: I don't think `__ufunc__` is a thing.

File: docs/construct.rst [hunk 2]
```diff
@@ -140,8 +140,31 @@ DOK arrays also support fancy indexing assignment if and only if all dimensions
...
+the following code will perform elementwise equality comparison on the two arrays
+and return a new boolean :obj:`DOK` array.
+
+.. code-block:: python
+
+   x = sparse.random((10, 10), 0.5, format="dok")
+   y = np.random.random((10, 10))
+   x == y
+
+:obj:`DOK` arrays are returned from elemwise functions and standard `__ufunc__` operations if and only if all
```
Comment (hameerabbasi) [inline docs/construct.rst:164]: Same here.

Comment (DragaDoncila) [reply docs/construct.rst]: I changed it to just say universal... I'm not sure if there's a better technical term to refer to it by

**Comment by hameerabbasi** [comment]
I think the technical term would be operators and ufuncs, without the double-underscore.

**Comment by DragaDoncila** [comment]
@hameerabbasi I've made that change. Is there anything else you'd like documented?

**Review by hameerabbasi — APPROVED** [review]

**Comment by hameerabbasi** [comment]
Thanks, @DragaDoncila!

**Pull request merged**